domain/src/main/java/org/oppia/domain/topic/StoryProgressController.kt

```diff
@@ -71,11 +283,6 @@ class StoryProgressController @Inject constructor(
     }
   }
 
-  // TODO(#21): Hide this functionality behind a data provider rather than punching a hole in this controller.
-  internal fun retrieveStoryProgress(storyId: String): StoryProgress {
-    return createStoryProgressSnapshot(storyId)
-  }
-
   private fun trackCompletedChapter(storyId: String, explorationId: String) {
```

Comment:
Make sure this is getting used or no anywhere else remove.

Reply:
+1 was this meant to be the replacement for ``recordCompletedChapter``? I suggest cleaning up & unneeded pieces of this controller now that the real implementation is in place.

Reply:
Actually this was need for some of the functions that I was thinking of implementing in this PR. But for now, I will remove this and introduce this again if needed in next PR.